### ISSUE TITLE
Need of flexible loss function GeneralLoss which can change behavior based on params

### ISSUE THREAD

**Issue by ashutosh1919** [tensorflow/addons#1646]
**Describe the feature and the current behavior/state.**  
Need of flexible loss function GeneralLoss which can change behavior based on params `alpha` and `scale`. Please take a look at [this paper](https://arxiv.org/pdf/1701.03077.pdf). 

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): yes
- Is there already an implementation in another framework? (if so, where): [google research](https://github.com/google-research/google-research/blob/master/robust_loss/general.py)
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**  
loss

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
